orttraining/orttraining/core/optimizer/megatron_transformer.h

```diff
@@ -75,6 +87,8 @@ class MegatronTransformer : public GraphTransformer {
   std::unordered_map<std::string, std::string>& updated_weight_names_;
   std::unordered_set<std::string>& weights_to_train_;
   std::unordered_map<std::string, training::TrainingSession::PartitionInfo>& weight_partition_info_;
+  training::TrainingSession::OptimizerState& initial_optimizer_states_;
```

Comment:
const ref?

Reply:
we are modifying the initial optimizer states passed: we partition the moments the same way as the weight is partitioned. Thus cannot be const